src/delta_lake/data_store.py

```diff
@@ -0,0 +1,203 @@
+    """
+    return json.dumps(data) if return_json else data
+
+
+def get_databases(
+    spark: Optional[SparkSession] = None,
+    use_postgres: bool = True,
+    return_json: bool = True,
+) -> Union[str, List[str]]:
+    """
+    Get the list of databases in the Hive metastore.
+
+    Args:
+        spark: Optional SparkSession to use (if use_postgres is False)
+        use_postgres: Whether to use PostgreSQL direct query (faster) or Spark
+        return_json: Whether to return JSON string or raw data
+
+    Returns:
+        List of database names, either as JSON string or raw list
+    """
+
+    def _get_dbs(session: SparkSession) -> List[str]:
+        return [db.name for db in session.catalog.listDatabases()]
+
+    if use_postgres:
+        databases = hive_metastore.get_databases()
+    else:
+        databases = _execute_with_spark(_get_dbs, spark)
+
+    return _format_output(databases, return_json)
+
+
+def get_tables(
+    database: str,
+    spark: Optional[SparkSession] = None,
+    use_postgres: bool = True,
+    return_json: bool = True,
+) -> Union[str, List[str]]:
+    """
+    Get the list of tables in a specific database.
+
+    Args:
+        database: Name of the database
+        spark: Optional SparkSession to use (if use_postgres is False)
+        use_postgres: Whether to use PostgreSQL direct query (faster) or Spark
+        return_json: Whether to return JSON string or raw data
+
+    Returns:
+        List of table names, either as JSON string or raw list
+    """
+
+    def _get_tbls(session: SparkSession, db: str) -> List[str]:
+        return [table.name for table in session.catalog.listTables(dbName=db)]
+
+    if use_postgres:
+        tables = hive_metastore.get_tables(database)
+    else:
+        tables = _execute_with_spark(_get_tbls, spark, database)
+
+    return _format_output(tables, return_json)
+
+
+def get_table_schema(
+    database: str,
+    table: str,
+    spark: Optional[SparkSession] = None,
+    return_json: bool = True,
+) -> Union[str, List[str]]:
+    """
+    Get the schema of a specific table in a database.
+
+    Args:
+        database: Name of the database
+        table: Name of the table
+        spark: Optional SparkSession to use
+        return_json: Whether to return JSON string or raw data
+
+    Returns:
+        List of column names, either as JSON string or raw list
+    """
+
+    def _get_schema(session: SparkSession, db: str, tbl: str) -> List[str]:
+        try:
+            return [
+                column.name
+                for column in session.catalog.listColumns(dbName=db, tableName=tbl)
+            ]
+        except Exception as e:
+            raise DeltaSchemaError(
+                f"Error retrieving schema for table {tbl} in database {db}: {e}"
+            )
+
+    columns = _execute_with_spark(_get_schema, spark, database, table)
+    return _format_output(columns, return_json)
+
+
+def get_db_structure(
+    spark: Optional[SparkSession] = None,
```

Comment:
introduced an additional `spark` arg comparing to original method:
https://github.com/kbase/cdm-jupyterhub/blob/main/src/spark/data_store.py#L115-L117